src/org/openj9/envInfo/MachineInfo.java

```diff
@@ -100,21 +115,88 @@ private String execCommands(String[] commands) {
 			proc.waitFor();
 		} catch (IOException | InterruptedException e) {
 			e.printStackTrace();
+			return "Command could not be executed";
 		}
 		return rt;
 	}
 
+	private String parseInfo(String version) {
+		Pattern pattern = Pattern.compile("[0-9]+[.][0-9]+([.][0-9]+)?"); 
+		Matcher matcher = pattern.matcher(version);
+		if (matcher.find()) {
+			return matcher.group(0);
+		} else {
+			return "";
+		}
+	}
+
+	private ArrayList<Integer> versionStr2ArrList(String version) throws NumberFormatException {
+		String[] versionSplit = version.split("\\.");
+		ArrayList<Integer> versionArr = new ArrayList<Integer>();
+		for (int i=0; i < versionSplit.length; i++) {
+			versionArr.add(Integer.parseInt(versionSplit[i]));
+		}
+		return versionArr;
+	}
+
+	private void makeSameLength(ArrayList<Integer> list, int requiredLength){
+		int differenceInLength = requiredLength - list.size();
+		for (int i=0; i<differenceInLength; i++){
+			list.add(0);
+		}
+	}
+
+	private boolean validateVersion(String versionName, String actualVersionStr, String requriedVersionStr) {
+		boolean isValid = true;
+		try { 
+			ArrayList<Integer> accVer = versionStr2ArrList(actualVersionStr);
+			ArrayList<Integer> reqVer = versionStr2ArrList(requriedVersionStr);
+			int accVerLen = accVer.size();
+			int reqVerLen = reqVer.size();
+			if (accVerLen > reqVerLen) {
+				makeSameLength(reqVer, accVerLen);
+			} else if (accVerLen < reqVerLen) {
+				makeSameLength(accVer, reqVerLen);
+			}
+			for (int i=0; i < accVer.size(); i++) {
+				if (reqVer.get(i) > accVer.get(i)){
+					isValid = false;
+				} else if (reqVer.get(i) < accVer.get(i)) {
+					break;
+				}
+			}
+			if (!isValid) {
+				System.out.println("Error: required " + versionName + ": " + requriedVersionStr + ". Installed version: " + actualVersionStr);
+			}
+		} catch (NumberFormatException e){
+			System.out.println("Warning: version information (" + versionName + " : " + requriedVersionStr + ") cannot be extracted.");	
```

Comment:
This message is misleading. The output from the Grinder:
```
Warning: version information (antVersion : 1.9.6) cannot be extracted.
```

The problem is that we cannot get the `actualVersionStr `, not `requriedVersionStr `. Please change it to:
```
System.out.println("Warning: versionName + " information cannot be extracted.");
System.out.println(versionName + " output: " + actualVersionStr);

```

Reply:
updated 👍 https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder_TKG/765/consoleFull